### ISSUE TITLE
Option to NOT replace comma

### ISSUE THREAD

**Issue by alexjamesbrown** [blakeembrey/title-case#3]
Sometimes, a title could have a comma -

But if i pass in:

'a title, with a comma'

I get back:

'A Title With A Comma'

**Comment by blakeembrey** [comment]
This isn't really the right module for that sort of thing, sorry. There's a lot more to go into a proper title library including things like lower-case minor words, supporting different punctuation, etc and it's pretty subjective. It's definitely possible to build a module for that though, but I'd have a look around.

**Comment by alexjamesbrown** [comment]
if a comma is in the lowercase version of the input', it's currently incorrectly stripping that out - thus, it's grammatically incorrect, and taking on a bit too much responsibility...

To Title Case

implies it will uppercase each word in the given input.

Nowhere does it state "Oh, and by the way, I'm gonna remove your commas and anything else I feel like"

**Comment by blakeembrey** [comment]
Currently it's stripping out all non-word characters. It's only for converting some other cased string to title-cased string, not necessarily turning it into a title. I guess it's a confusing name in that regard. You're going to also have issues with converting any other case, such as dash case or camel case so this is probably a documentation issue. To create what you're after though, it is fairly straightforward until you hit the opinionated areas and odd grammar areas (especially non-ascii).

**Comment by alexjamesbrown** [comment]
Yeah, I agree it's a documentation issue... the readme implies I get back what I put in, just with case alterations - no mention of it stripping out non-word chars

**Comment by blakeembrey** [comment]
Here's something on the topic that might interest you: http://ejohn.org/blog/title-capitalization-in-javascript/

I'll keep this issue open to make a note of updating the documentation. If this is something of interest, I'll look at writing an improved version but mostly I've found in the projects I've written that it becomes tricky with lots of custom rules and sometimes it's not even worth it (e.g. with user submitted content)

**Comment by jimkang** [comment]
change-case in general is pretty great, but I just thought I'd note that I found this really confusing as well. Only after looking through the code did I realize this was probably intentional.

**Comment by alerque** [comment]
I really think this problem is being approached all wrong. I understand this module cannot be expected to make full titles because of all the grammar complexities and possible style guides involved for each language. I agree this is the wrong place for that. _So don't even try._ This issue here is not that it's not doing enough, its that it's doing too much.

This module is tripping us up because it's doing something beyond the scope of the function. The solution is to do less. Don't do  anything but change the _CASE_ of letters. I can take a string and make it all caps or all lower case and back and fourth round trip without  breaking anything. The title-case function currently destroys data that has nothing to do with casing. It should be less aggressive and only change things that it knows about without venturing into doing random things to characters that aren't relevant to the function.

**Comment by blakeembrey** [comment]
It's not really doing to much, it wasn't designed for this at all. If you wish to make the module do something else, submit a PR and I can accept it. But don't put the blame on misunderstanding - the README shows examples of switching between cases (which all the `*-case` modules do) - https://github.com/blakeembrey/title-case#usage. It can be better documented, and I'm happy to make the change, but stuff like this isn't useful. I fail to see how you believe writing a correct title casing algorithm that suits everyones needs and different languages is "doing less", but I would like to support it as the current module is poorly named and utilised.

**Comment by jimkang** [comment]
For people that do want the comma-preserving behavior, the [titlecase](https://www.npmjs.com/package/titlecase) module does this.

**Comment by alexjamesbrown** [comment]
That's the one ended up using. Does exactly what it says on the tin.
